opengrok-indexer/src/main/java/org/opengrok/indexer/index/IndexDatabase.java

```diff
@@ -505,6 +534,17 @@ public void update() throws IOException {
                         }
                     }
 
+                    /*
+                     * Since we are avoiding an IndexAnalysisSettings bump to
+                     * store needed state related to #Lines/LOC, we need to
+                     * ensure that at least one D-document gets saved. Record
+                     * zeroes for a .foo under sourceRoot to do this.
+                     */
+                    countsAggregator.register(new NumLinesLOC("/.foo", 0, 0));
```

Comment:
I'm not sure if I follow this logic. Could you extract the filename to a constant?

Reply:
I don't understand how a constant will help anyone follow if that long comment cannot do.

Reply:
These two weren't connected.

Reply:
Oh. OK I extracted a constant and updated the comment hopefully for clarity.

Reply:
Understood, nice.